sockeye/decoder.py

```diff
@@ -158,7 +162,15 @@ def __init__(self,
             self.mapped_context_b = mx.sym.Variable("%smapped_context_bias" % prefix)
 
         # Stacked RNN
-        self.rnn = rnn.get_stacked_rnn(self.rnn_config, self.prefix)
+        if self.rnn_config.num_layers == 1 or not self.attention_in_upper_layers:
+            self.attention_rnn = rnn.get_stacked_rnn(self.rnn_config, self.prefix, False)
```

Comment:
I was considering "first_layer_rnn" or something similar. In the end I found clearer as attention_rnn, but I don't have a strong argument for it.